### PR TITLE
feat: add Topology field on ServerConfig model

### PR DESCRIPTION
**Pull request by lucasponce**

### CODE REVIEW

**Comment by andreadecorte** [comment]
we should also add 
ProvisionShardDedicatedTopology = "dedicated"

File: model/clusters_mgmt/v1/server_config_type.model [hunk 1]
```diff
@@ -21,4 +21,7 @@ class ServerConfig {
 
 	// The kubeconfig of the server
 	Kubeconfig String
+
+	// The topology of a provision shard (Optional)
```
Comment (gdbranco) [inline model/clusters_mgmt/v1/server_config_type.model:25]: Not sure why nothing on this file has the '.', but we usually required it. Do you know if there's some reason for this file not to have that for the comments?

**Review by gdbranco — APPROVED** [review]

File: model/clusters_mgmt/v1/server_config_type.model (hunk 1 above)
Comment (lucasponce) [reply model/clusters_mgmt/v1/server_config_type.model:25]: Ah, true, I've created the file from scratch, and I forgot.

I'll update it

**Review by tbrisker — APPROVED** [review]

**Pull request merged**